runtime/allocator.cpp

```diff
@@ -306,6 +282,48 @@ MemoryReplacementGuard::~MemoryReplacementGuard() {
 
 } // namespace dl
 
+// sanitizers aren't happy with custom realization of malloc-like functions
+#if !ASAN_ENABLED and !defined(__APPLE__)
+
+extern "C" void *__libc_malloc(size_t size);
+extern "C" void *__libc_calloc(size_t nmemb, size_t size);
+extern "C" void *__libc_realloc(void *mem, std::size_t new_size);
+extern "C" void __libc_free(void *mem);
+extern "C" void *__libc_memalign(size_t alignment, size_t size);
+
+void *malloc(size_t size) {
+  return dl::is_malloc_replaced() ? dl::script_allocator_malloc(size) : __libc_malloc(size);
+}
+
+void *calloc(size_t nmemb, size_t size) {
+  return dl::is_malloc_replaced() ? dl::script_allocator_calloc(nmemb, size) : __libc_calloc(nmemb, size);
+}
```

Comment:
We didn't replace calloc before, but now it's replaced. Is this intentional?

Reply:
Yes it is. `calloc` under the hood uses `__malloc_hook`, substituted in our previous implementation. Now we ought to override `calloc` if we want to preserve the same behaviour.